src/Testcontainers/Builders/TestcontainersBuilderAzuriteExtension.cs

```diff
@@ -0,0 +1,139 @@
+  using System.IO;
+  using DotNet.Testcontainers.Configurations.Modules.Databases;
+  using DotNet.Testcontainers.Containers;
+  using JetBrains.Annotations;
+
+  /// <summary>
+  /// This class applies the extended Testcontainer configurations for Azurite.
+  /// </summary>
+  [PublicAPI]
+  public static class TestcontainersBuilderAzuriteExtension
+  {
+    public static ITestcontainersBuilder<AzuriteTestcontainer> WithAzurite(this ITestcontainersBuilder<AzuriteTestcontainer> builder, AzuriteTestcontainerConfiguration configuration)
+    {
+      builder = builder
+        .WithImage(configuration.Image)
+        .WithEnvironment(new ReadOnlyDictionary<string, string>(configuration.Environments))
+        .WithOutputConsumer(configuration.OutputConsumer)
+        .WithWaitStrategy(configuration.WaitStrategy)
+        .ConfigureContainer(container =>
+        {
+          container.ContainerBlobPort = configuration.RunBlobOnly || configuration.RunAllServices ? AzuriteTestcontainerConfiguration.DefaultBlobPort : 0;
+          container.ContainerQueuePort = configuration.RunQueueOnly || configuration.RunAllServices ? AzuriteTestcontainerConfiguration.DefaultQueuePort : 0;
+          container.ContainerTablePort = configuration.RunTableOnly || configuration.RunAllServices ? AzuriteTestcontainerConfiguration.DefaultTablePort : 0;
+        });
+
+      if (configuration.RunBlobOnly || configuration.RunAllServices)
+      {
+        builder = builder.WithPortBinding(configuration.BlobPort, AzuriteTestcontainerConfiguration.DefaultBlobPort);
+      }
+
+      if (configuration.RunQueueOnly || configuration.RunAllServices)
+      {
+        builder = builder.WithPortBinding(configuration.QueuePort, AzuriteTestcontainerConfiguration.DefaultQueuePort);
+      }
+
+      if (configuration.RunTableOnly || configuration.RunAllServices)
+      {
+        builder = builder.WithPortBinding(configuration.TablePort, AzuriteTestcontainerConfiguration.DefaultTablePort);
+      }
+
+      if (!string.IsNullOrWhiteSpace(configuration.Location))
+      {
+        builder = builder.WithBindMount(configuration.Location, AzuriteTestcontainerConfiguration.DefaultLocation);
+      }
+
+      builder = builder
+        .WithCommand(GetMainCommand(configuration))
+        .WithCommand(GetServiceEndpointArgs(configuration))
+        .WithCommand(GetWorkspaceLocation())
+        .WithCommand(GetSilentMode(configuration))
+        .WithCommand(GetLooseMode(configuration))
+        .WithCommand(GetSkipApiVersionCheck(configuration))
+        .WithCommand(GetDisableProductStyleUrl(configuration))
+        .WithCommand(GetDebugLog(configuration));
+
+      return builder;
+    }
+
+    private static string[] GetDebugLog(AzuriteTestcontainerConfiguration configuration)
+    {
+      var debugLogFilePath = Path.Combine(AzuriteTestcontainerConfiguration.DefaultLocation, "debug.log");
+      return configuration.DisableProductStyleUrl ? new[] { "--debug", debugLogFilePath } : null;
```

Comment:
Shouldn`t this be `configuration.Debug`?.

Reply:
Good catch. Should be each option tested?
Fixed in [bddb8be](https://github.com/testcontainers/testcontainers-dotnet/pull/555/commits/bddb8be4dc91b51a89a704799bbe006d784e4f8c)

Reply:
Not necessarily, but if we can cover more or all it would be great. Usually, I try to cover all.